### ISSUE TITLE
Duplicate AttendanceRecord entries sometimes get created

### ISSUE THREAD

**Issue by nharper** [nharper/allegro#17]
The AttendanceRecord model has a validation on it `validates_uniqueness_of :performer, :scope => :rehearsal` with the intention of making sure there is at most one row for a (performer_id, rehearsal_id) pair. Running this SQL query: `SELECT c, performer_id, rehearsal_id FROM (SELECT COUNT(*) AS c, performer_id, rehearsal_id FROM attendance_records GROUP BY performer_id, rehearsal_id) AS t WHERE c > 1` should return 0 rows, but it does not.

Most likely, there is a race condition in RehearsalsController lines 158-164. My hypothesis is that two requests coming in (to update data for the same rehearsal at the same time) hit the `AttendanceRecord.where` statement, and both make new record objects. When the records get saved, each request is trying to insert a new row, and the second request to insert a new row for a particular (performer_id, rehearsal_id) pair fails the validation.

**Comment by nharper** [comment]
With any luck, 7b0121ff59f329c8851b176e7fa758d786b37544 will fix this issue.

**Comment by nharper** [comment]
Rolled back 7b0121f because it produced the following error in the server log:

```
App 21928 stderr: Completed 500 Internal Server Error in 6ms
App 21928 stderr:
App 21928 stderr: NoMethodError (undefined method `transaction' for ActiveRecord:Module):
App 21928 stderr:   app/controllers/rehearsals_controller.rb:158:in `reconcile'
App 21928 stderr:
App 21928 stderr:
```

**Comment by nharper** [comment]
8c247b7ad67fe14bbd4d345b17aaf80794215d8a fixed the bug from 7b0121f, and the SQL query in the description returns 0 rows right now. I'll leave this open for a few weeks to check on it.

**Comment by nharper** [comment]
I think this issue is resolved.
